### PR TITLE
Add inc fp8 qunatization documentation

### PR DESCRIPTION
**Pull request by nirda7**

### CODE REVIEW

**Review by kzawora-intel — DISMISSED** [review]

File: docs/source/quantization/inc.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+FP8 INC
+==================
+
+vLLM supports FP8 (8-bit floating point) weight and activation quantization using INC (Intel Neural Compressor) on hardware acceleration of Intel Gaudi (HPU).
+Currently, only Llama models quntization are supported.
+
+Please visit the Intel Gaudi documentation of `Run Inference Using FP8  <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html>`_.
+
+In order to run Inference it is required to have Measurements/Scales files:
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:11]: Done

File: docs/source/quantization/inc.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    llm.llm_engine.model_executor.shutdown()
+
+.. note::
+
+   Make sure to supply the "QUANT_CONFIG" enviornment variable which points to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ with MEASURE mode.
+
+Run Inference Using FP8
+-----------------------
+
+Inte Gaudi supports quantization of Linear Layers, KV-Cache and functions like Matmul and Softamx as shown in:
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+`Supported Modules <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-modules>`_.
+`Supported Functions <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-functions>`_.
+
+In order to run Inference it requies to have Scales which located in scale files according to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ dump_stats_path.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    llm = LLM("llama3.1/Meta-Llama-3.1-8B-Instruct", quantization="inc")
+    ...
+    # Call llm.generate on the required prompts and sampling params.
+    ...
+    llm.llm_engine.model_executor.shutdown_inc()
+    llm.llm_engine.model_executor.shutdown()
+
+.. note::
+
+   Make sure to supply the "QUANT_CONFIG" enviornment variable which points to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ with MEASURE mode.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
+.. _INC:
+
+FP8 INC
+==================
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:4]: Done

File: docs/source/quantization/inc.rst [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+Please visit the Intel Gaudi documentation of `Run Inference Using FP8  <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html>`_.
+
+In order to run Inference it is required to have Measurements/Scales files:
+
+Retrieve Measurements
+---------------------
+
+To obtain measurement files:
+* Use the "inc" quantization method (as parameter to the LLM object).
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:17]: Done

File: docs/source/quantization/inc.rst [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+* Call shutdown_inc and shutdown methods of the model_executor in the end of the run.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:18]: Done

File: docs/source/quantization/inc.rst [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+Inte Gaudi supports quantization of Linear Layers, KV-Cache and functions like Matmul and Softamx as shown in:
+`Supported Modules <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-modules>`_.
+`Supported Functions <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-functions>`_.
+
+In order to run Inference it requies to have Scales which located in scale files according to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ dump_stats_path.
+If none exist they can be generated during inference run using the measurement files (should be loacted in the same folder).
+
+To run inference (and obtain scale files):
+* Use the "inc" quantization method (as parameter to the LLM object).
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:45]: Done

File: docs/source/quantization/inc.rst [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+* Use the "fp8_inc" kv cache dtype (as parameter to the LLM object).
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:46]: Done

File: docs/source/quantization/inc.rst [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+* Call shutdown method of the model_executor in the end of the run.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:47]: Done

File: docs/source/quantization/inc.rst [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    from vllm import LLM
+    llm = LLM("llama3.1/Meta-Llama-3.1-8B-Instruct", quantization="inc", kv_cache_dtype="fp8_inc")
+    ...
+    # Call llm.generate on the required prompts and sampling params.
+    ...
+    llm.llm_engine.model_executor.shutdown()
+
+.. note::
+
+    Make sure to supply the "QUANT_CONFIG" environment variable which points to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ with QUANTIZE mode.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:60]: Done

File: docs/source/quantization/inc.rst [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    llm.llm_engine.model_executor.shutdown()
+
+.. note::
+
+    Make sure to supply the "QUANT_CONFIG" enviornment variable which points to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ with QUANTIZE mode.
+
+Specifying Device for the Model's Weights Uploading
+---------------------------------------------------
+
+It is possible to upload the (unquantized) weights on a different device before qunantizing them
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:65]: Done

File: docs/source/quantization/inc.rst [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+.. note::
+
+    Make sure to supply the "QUANT_CONFIG" enviornment variable which points to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ with QUANTIZE mode.
+
+Specifying Device for the Model's Weights Uploading
+---------------------------------------------------
+
+It is possible to upload the (unquantized) weights on a different device before qunantizing them 
+and moving to the device on which the model will run.
+Use the weights_load_device parameter for the LLM object to specify this device.
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst:67]: Done

File: docs/source/quantization/inc.rst [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+Run Inference Using FP8
+-----------------------
+
+Inte Gaudi supports quantization of Linear Layers, KV-Cache and functions like Matmul and Softamx as shown in:
+`Supported Modules <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-modules>`_.
+`Supported Functions <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-functions>`_.
+
+In order to run Inference it requies to have Scales which located in scale files according to the `Json config file <https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html#supported-json-config-file-options>`_ dump_stats_path.
+If none exist they can be generated during inference run using the measurement files (should be loacted in the same folder).
```
[the remark this answers is not in the document]
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+Once you've completed the model calibration process and collected the measurements, you can run FP8 inference with vLLM using the following command:
+
+.. code-block:: bash
+
+    export QUANT_CONFIG=/path/to/quant/config/inc/meta-llama-3.1-405b-instruct/maxabs_measure_g3.json
+    vllm serve meta-llama/Llama-3.1-405B-Instruct --quantization inc --kv-cache-dtype fp8_inc --weights-load-device cpu --tensor_paralel_size 8
+
+.. tip::
+    If you are just prototyping or testing your model with FP8, you can use the ``VLLM_SKIP_WARMUP=true`` environment variable to disable the warmup stage, which can take a long time. However, we do not recommend disabling this feature in production environments, as it causes a dramatic performance drop.
```
Comment (auvarovahabana) [inline docs/source/quantization/inc.rst:30]: However, we do not recommend disabling this feature in production environments as it causes a significant performance drop.

File: docs/source/quantization/inc.rst [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+
+.. tip::
+    When using FP8 models, you may experience timeouts caused by the long compilation time of FP8 operations. To mitigate this problem, you can use these two environment variables:
```
Comment (auvarovahabana) [inline docs/source/quantization/inc.rst:33]: To mitigate this problem, you can use the below environment variables:

File: docs/source/quantization/inc.rst [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+    # Call llm.generate on the required prompts and sampling params.
+    ...
+    llm.llm_engine.model_executor.shutdown()
+
+Specifying Device for the Model's Weights Uploading
+---------------------------------------------------
+
+It is possible to load the unquantized weights on a different device before quantizing them, then moving them to the device on which the model will run.
+This reduces the device memory footprint of model weights, as only quantized weights are stored in device memory.
+To set the load device, use the ``weights_load_device`` parameter for the ``LLM`` object, or ``--weights-load-device`` command line parameter in online mode.
```
Comment (auvarovahabana) [inline docs/source/quantization/inc.rst:59]: To set the device to upload weights, use the ``weights_load_device`` parameter for the ``LLM`` object, or ``--weights-load-device`` command line parameter when running online inference:

File: docs/source/quantization/inc.rst (hunk 15 above)
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst (hunk 16 above)
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

File: docs/source/quantization/inc.rst (hunk 17 above)
Comment (nirda7) [reply docs/source/quantization/inc.rst]: Done

**Review by auvarovahabana — APPROVED** [review]

File: docs/source/quantization/inc.rst [hunk 18]
```diff
@@ -0,0 +1,64 @@
+.. _INC:
+
+FP8 INC
+=======
+
+vLLM supports FP8 (8-bit floating point) weight and activation quantization using Intel® Neural Compressor (INC) on Intel® Gaudi® 2 and Intel® Gaudi® 3 AI accelerators.
+Currently, quantization is supported only for Llama models.
```
Comment (thuang6) [inline docs/source/quantization/inc.rst:7]: is it better to revise to "quantization is validated only in Llama models"?  FP8 quantization should support more models though not verified widely and may has issue in some models. Better not discourage user try with other models.

Comment (nirda7) [reply docs/source/quantization/inc.rst:7]: The problem with other models is that there should be some modification/preparation so we can run full quantization on them - only the linear layers will be quantized for now, but for the matmuls, softmaxes, kv-caches .... we will have to do the nn.module wrapping in order to quantize them...
as can be seen here: (supported modules, supported functions, custom patched modules)
https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Quantization/Inference_Using_FP8.html#supported-modules

Comment (thuang6) [reply docs/source/quantization/inc.rst]: we tried Qwen model and matmul, softmax, kv-cache are quantized to FP8.

Comment (nirda7) [reply docs/source/quantization/inc.rst]: If it uses the same paged attention algorithm (flat_pa used for llama) - so yes - everything we quantized there for llama will also be quantized for other models that use this PA logic.

**Review by auvarovahabana — APPROVED** [review]

**Review by afierka-intel — APPROVED** [review]
Read all and looks good for me.

**Review by thuang6 — APPROVED** [review]
LGTM

**Review by kzawora-intel — APPROVED** [review]
